.github/workflows/rust.yml

```diff
@@ -1,18 +1,60 @@
 name: Rust
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ main ]
```

Comment:
My repo still works using the `master` branch (this is the default behavior of Jetbrains' IntelliJ.

Otherwise nice idea!

Reply:
oh, oops, yes, you do name the primary as `master`, my mistake.  BTW, it is generally recommended to rename it to `main` -- e.g. see [here](https://github.com/github/renaming).  I would add both so that it makes it easier to just rename when you are ready.